### DISCUSSION TITLE
Create thumbnails of 100% of the canvas

### DISCUSSION THREAD

**Discussion by CamboimGabriel** [xyflow/xyflow#2717] [Q&A] [answered]
Hi guys, I'm trying to generate a thumbnail of the Canvas ( for our Dashboard )  and I'm using the example of react-flow to generate html to png image, but the problem is that the generated screenshot is not fully showing the full content of the canvas, it only captures what is on the user's screen at the moment - which if they are zooming at the time is not ideal! Do you guys have any idea to create a workaround for that?

**Comment by moklick** [comment]
I think you have two options:

1. you can remember the current viewport, for the view, take a screenshot and restore the previous viewport
2. create another hidden pane with a clone of your flow and take a screenshot from that one

**Reply by CamboimGabriel** [reply]
The first one will provide a weird visual for the user, since I just can't change viewport and restore instantly, so, the second one seems the best way to do it, but, how can I do this?

**Reply by moklick** [reply]
This is how you could do it (roughly):

1. create temporary div with `createElement`
2. copy flow with `cloneNode`
3. add cloned flow to the temp div with `appendChild` 
4. do screenshot
5. remove temp div

**Reply by CamboimGabriel** [reply]
Awesome! But how can I fit view only this temp div?

**Comment by moklick** [comment, marked as answer]
You can pass a `style` option `{ transforn: 'translate(0,0) scale(1)' }` and only select the viewport: 

```js
toPng(document.querySelector(".react-flow__viewport"), {
  style: {
    transform: "translate(0, 0) scale(1)"
  }
})
```

https://codesandbox.io/s/image-fullscreen-936g03?file=/DownloadButton.js:371-417
